sympy/stats/rv_interface.py

```diff
@@ -339,6 +340,61 @@ def factorial_moment(X, n, condition=None, **kwargs):
     """
     return expectation(FallingFactorial(X, n), condition=condition, **kwargs)
 
+def median(X, evaluate=True, **kwargs):
+    r"""
+    Calculuates the median of the probability distribution.
+    Mathematically, median of Probability distribution is defined as all those
+    values of `m` for which the following condition is satisfied
+
+    .. math::
+        P(X\geq m)\geq 1/2 \hspace{5} \text{and} \hspace{5} P(X\leq m)\geq 1/2
+
+    Parameters
+    ==========
+
+    X: The random expression whose median is to be calculated.
+
+    Returns
+    =======
+
+    The FiniteSet or an Interval which contains the median of the
+    random expression.
+
+    Examples
+    ========
+
+    >>> from sympy.stats import Normal, Die, median
+    >>> N = Normal('N', 3, 1)
+    >>> median(N)
+    FiniteSet(3)
+    >>> D = Die('D')
+    >>> median(D)
+    FiniteSet(3, 4)
```

Comment:
If this returns the set of all possible medians then this should be `Interval(3, 4)`

Reply:
No, it does not refer to all such values, the median are only `3` and `4` as its set is `FiniteSet(1, 2, 3, 4, 5, 6)` so from this only `3, 4` are median.

Reply:
@oscarbenjamin  Is it good now?